content/python/concepts/functions/functions.md

```diff
@@ -184,15 +237,24 @@ add = lambda a, b: a + b
 
 The expression to the right of the assignment operator is called a "lambda expression". The Python interpreter takes this expression and defines a function object which can be bound to an identifier (in this case, `add`). There is no difference between binding a function to a name using the assignment operator or by using the `def` keyword.
 
+Parameters are optional when defining an anonymous function. However, a function body must be present, and it must only contain a single return expression.
+
 Lambda expressions are most commonly used to define simple, single-use functions which are passed to higher-order functions.
 
 ```py
 numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 
 odd_numbers = filter(lambda n : n % 2 == 1, numbers)
 
-print(list(perfect_squares))
+print(list(odd_numbers))
 # Output: [1, 3, 5, 7, 9]
 ```
 
+Anonymous functions can also be invoked immediately after they are defined.
+
+```py
+print((lambda a, b: a + b)(1986, 33))
+# output: 2019
+```
+
 For more complex functions with larger bodies, it is recommended to use the `def` keyword to keep code readable.
```

Comment:
If the function can be written a single line, `lambda` could be useful. Otherwise, if the function is more complex and spans multiple lines, it is recommended to use the `def` keyword.